src/DonationForms/DonationFormsAdminPage.php

```diff
@@ -49,6 +60,30 @@ public function loadScripts()
         );
     }
 
+    /**
+     * Make REST request to Donation Forms endpoint before page load
+     * @unreleased
+     */
+    public function preloadForms()
```

Comment:
I'm not sure I'm a fan of this approach - running additional HTTP requests from the backend. 

What do you think about we pull this data from the repository? We can also create a `ViewModel` class for donation forms and use it here and in the `ListForms` endpoint. I know this would be extra work to do almost the same thing, but in the end, it will run much faster. In addition to `useFallbackAsInitial` middleware, this would be a win-win situation.

Reply:
I will finish backend tasks for Scope 3 soon, so I can give you a hand here.

Reply:
@alaca As far as I understand, the `preloadForms` method currently doesn't do an HTTP request, either on the client or the server. It creates a mock `WP_REST_Request` object, then passes it to `rest_do_request`. The REST server matches the request to an endpoint, and then calls the endpoint's handler function using `call_user_func` and returns the response object. So we're just passing around objects and calling functions on them, without getting HTTP involved. Like this: https://wpscholar.com/blog/internal-wp-rest-api-calls/

Reply:
Great, I didn't know that. As long as we don't make HTTP requests, I'm fine with that.